src/jit/lclvars.cpp

```diff
@@ -3985,47 +4002,68 @@ void Compiler::lvaMarkLocalVars()
         }
     }
 
-    BasicBlock* block;
-
-#ifndef DEBUG
-    // Assign slot numbers to all variables.
-    // If compiler generated local variables, slot numbers will be
-    // invalid (out of range of info.compVarScopes).
+    // Ref counting is now enabled normally.
+    lvaRefCountState = RCS_NORMAL;
 
-    // Also have to check if variable was not reallocated to another
-    // slot in which case we have to register the original slot #.
+#if defined(DEBUG)
+    const bool setSlotNumbers = true;
+#else
+    const bool setSlotNumbers = opts.compScopeInfo && (info.compVarScopesCount > 0);
+#endif // defined(DEBUG)
 
-    // We don't need to do this for IL, but this keeps lvSlotNum consistent.
+    unsigned   lclNum = 0;
+    LclVarDsc* varDsc = nullptr;
 
-    if (opts.compScopeInfo && (info.compVarScopesCount > 0))
-#endif
+    // Fast path for minopts and debug codegen.
+    //
+    // Mark all locals as implicitly referenced and untracked.
+    // Don't bother sorting.
+    if (opts.MinOpts() || opts.compDbgCode)
```

Comment:
We probably could use a helper method that checked these two.

Reply:
Yeah, will put this on my todo list -- in particular there are places that just check for MinOpts which seems odd.